internal/docker/client.go

```diff
@@ -93,35 +96,44 @@ type Cli struct {
 	*client.Client
 	supportsBuildkit bool
 
+	env []string
+
 	creds     dockerCreds
 	initError error
 	initDone  chan bool
 }
 
-func ProvideDockerEnv(ctx context.Context, env k8s.Env, runtime container.Runtime) (DockerEnv, error) {
-	if runtime == container.RuntimeDocker {
-		if env == k8s.EnvMinikube {
-			envMap, err := minikube.DockerEnv(ctx)
-			if err != nil {
-				return nil, errors.Wrap(err, "ProvideDockerEnv")
-			}
-
-			return func(key string) string { return envMap[key] }, nil
-		} else if env == k8s.EnvMicroK8s {
-			return func(key string) string {
+func ProvideEnv(ctx context.Context, env k8s.Env, runtime container.Runtime) (Env, error) {
+	// TODO(dbentley): why do we allow DOCKER_HOST to be shadowed for microK8s but not for minikube?	if runtime == container.RuntimeDocker {
+	if env == k8s.EnvMinikube {
+		envMap, err := minikube.DockerEnv(ctx)
+		if err != nil {
+			return Env{}, errors.Wrap(err, "ProvideDockerEnv")
+		}
+
+		var envList []string
+		for k, v := range envMap {
+			envList = append(envList, fmt.Sprintf("%s=%s", k, v))
+		}
+
+		return Env{func(key string) string { return envMap[key] }, envList}, nil
+	} else if env == k8s.EnvMicroK8s {
+		return Env{
+			func(key string) string {
 				val := os.Getenv(key)
 				if val == "" && key == "DOCKER_HOST" {
 					return microK8sDockerHost
 				}
 				return val
-			}, nil
-		}
+			},
+			[]string{"DOCKER_HOST=" + microK8sDockerHost},
```

Comment:
it seems kinda weird that the func returns stuff from `os.Getenv` but the `[]string` doesn't. Maybe it'd make sense to just throw all of `os.Getenv` in there?

(also, unrelated to this PR, but do you know if it's intentional that `os.Getenv` isn't used in minikube's func?)

Reply:
Build Injections is what needs to be injected in addition to the current environment.

No idea about the os.Getenv use/non-use (and added a TODO above)